pipenv/project.py

```diff
@@ -353,7 +353,7 @@ def _pipfile(self):
         # mutation time!
         self.clear_pipfile_cache()
         for section in ('packages', 'dev-packages'):
-            p_section = dict(pfile.get(section, {}))
+            p_section = pfile.get(section, {})
             for key in list(p_section.keys()):
                 # Normalize key name to PEP 423.
                 norm_key = pep423_name(key)
```

Comment:
After I looked into the `contoml/file.py`, I find that using `pop` here won't break anything. Thanks anyway. New modification is committed.

Reply:
oh @frostming sorry it took me so long to see this here -- with this change, does pipenv properly normalize the pipfile again?

Reply:
According to my search result of the code base, normalization will happen when Pipfile is changed: add package/remove package/add index

Reply:
this might cause all comments in the file to be wiped out and re-normalize all names in the Pipfile (but hopefully we can add a unit test to check that doesn't happen)

Reply:
@jtratner @frostming this is not actually desired behavior -- we don't want to rewrite the case of the entire pipfile every time someone installs a single package, we only want to normalize the one they are installing

Reply:
Yup, so would you mind I change the behavior here, together with recasing issue #1855 ?